pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensionresources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/apis/extensions/validation"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/admission/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// WebhookPath is the HTTP handler path for this admission webhook handler.
+const WebhookPath = "/webhooks/validate-extension-resources"
+
+var gvk = schema.GroupVersionKind{
+	Group:   extensionsv1alpha1.SchemeGroupVersion.Group,
+	Version: extensionsv1alpha1.SchemeGroupVersion.Version,
+	Kind:    "ValidatingWebhookForExternalResources",
+}
+
+// New creates a new webhook handler validating CREATE and UPDATE requests for extension resources.
+func New(logger logr.Logger) *handler {
+	h := handler{
+		logger:    logger,
+		artifacts: make(map[metav1.GroupVersionResource]artifact),
+	}
+
+	h.addResource(metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "backupbuckets"},
+		artifact{
+			newEntity: func() client.Object { return new(extensionsv1alpha1.BackupBucket) },
+			validateResource: func(n, _ client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.BackupBucket)
+				return validation.ValidateBackupBucket(new)
+			},
+			validateResourceUpdate: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.BackupBucket)
+				old := o.(*extensionsv1alpha1.BackupBucket)
+				return validation.ValidateBackupBucketUpdate(new, old)
+			}})
+
+	h.addResource(metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "backupentries"},
```

Comment:
The bastions seems to be missing in this list and there is a already existing validation about them under `pkg/apis/extensions/validation/bastion.go`. Any concrete reason to do not include them?

Reply:
It is forgotten, will add it. Do you think it is wise to be added also in `pkg/seedadmissioncontroller/webhooks/admission/extensioncrds/admission.go` (because it is missing there too)?

Reply:
Good question. I am not sure whether the bastions need a deletion confirmation. @petersutter @rfranzke WDYT? We can leave this potential change for another PR as it is out of context of this PR.

Reply:
I think we decided that `Bastion`s don't need such protection (as they are ephemeral and only used for debugging purposes). There is no strong reason not to add it, though, I'd rather do it in a consistent way afterwards/separate PR.

Reply:
Make sense, won't touch `pkg/seedadmissioncontroller/webhooks/admission/extensioncrds/admission.go` in this PR.